tests/ui/arc_with_non_send_sync.rs

```diff
@@ -0,0 +1,12 @@
+#![warn(clippy::arc_with_non_send_sync)]
+#![allow(unused_variables)]
+use std::cell::RefCell;
+use std::sync::{Arc, Mutex};
+
+fn main() {
+    // This is safe, as `i32` implements `Send` and `Sync`.
+    let a = Arc::new(42);
```

Comment:
issue: we should also test that it does not lint in generic contexts

Reply:
Just to make sure I understand - would something like this cover your concerns?

```
mod unused {
    use std::cell::RefCell;
    struct Arc<T>(T);

    impl<T> Arc<T> {
        fn new(arg: T) -> Self {
            Arc(arg)
        }
    }

    fn no_warning() {
        // This is fine, since it's not the lang item Arc.
        let a = Arc::new(RefCell::new(42));
    }
}
```

Reply:
No, I meant checking that 

```
fn foo<T>(x: T) {
  let a = Arc::new(x)
}
```

shouldn't lint (I guess it's fine for it to lint because there's no `T: Sync`, but if you're making that lint then you should ensure it doesn't lint when `T: Sync` is specified)

Reply:
Ah - yeah, it's not handling that case properly. Let me try to tackle that (so that T: sync at least works.)

Reply:
I think it's best to not lint at all in the generic case; and add tests to ensure it doesn't.